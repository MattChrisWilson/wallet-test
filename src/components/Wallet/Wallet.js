import React, { Fragment } from 'react';

// @Flow

type Props = {
  loading: boolean,
  value: number,
};

const Wallet = (props: Props) => (props.loading) ? 
  <div>Please wait</div> : 
  <div>
    &pound;{props.value.toFixed(2)} 
    {
      props.onClick ? 
      <Fragment>
        <button onClick={() => props.onClick({ value: props.value, action: 'inc' })}>+</button>
        <button onClick={() => props.onClick({ value: props.value, action: 'dec' })}>-</button>
      </Fragment> :
      ''
    }
  </div>;

Wallet.defaultProps = {
	loading: false,
	value: 0,
}

export default Wallet;