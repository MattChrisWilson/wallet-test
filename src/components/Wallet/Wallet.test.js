import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Wallet from './Wallet';

configure({ adapter: new Adapter() });

const setup = (setupProps = {}) => {
	const defaultProps = {
		loading: false,
		value: 0,
	};

	const props = { ...defaultProps, ...setupProps };

	const comp = shallow(
		<Wallet loading={props.loading} value={props.value} onClick={props.onClick} />
	);

	return { props, comp };
}

describe('Wallet', () => {
  test('Renders without crashing', () => {
    const { comp } = setup();
    expect(comp).toMatchSnapshot();
  });

  test('Renders loading message when loading is true', () => {
    const { comp } = setup({ loading: true });
    expect(comp).toMatchSnapshot();
  });

  test('Renders value when passed', () => {
    const { comp } = setup({ value: 5 });
    expect(comp).toMatchSnapshot();
  });

  test('Renders buttons when onClick is passed', () => {
    const { comp } = setup({ onClick: jest.fn() });
    expect(comp).toMatchSnapshot();
  });

  test('calls onClick() on click', () => {
    const { props, comp } = setup({ onClick: jest.fn() });

    const link = comp.find('button').at(0);
    link.simulate('click');

    expect(props.onClick).toBeCalled();
  });
});