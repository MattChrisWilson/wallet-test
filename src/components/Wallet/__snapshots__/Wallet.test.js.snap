// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wallet Renders buttons when onClick is passed 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Wallet
    loading={false}
    onClick={[MockFunction]}
    value={0}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        "£",
        "0.00",
        <UNDEFINED>
          <button
            onClick={[Function]}
          >
            +
          </button>
          <button
            onClick={[Function]}
          >
            -
          </button>
        </UNDEFINED>,
      ],
    },
    "ref": null,
    "rendered": Array [
      "£",
      "0.00",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <button
              onClick={[Function]}
            >
              +
            </button>,
            <button
              onClick={[Function]}
            >
              -
            </button>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "inc-button",
            "nodeType": "host",
            "props": Object {
              "children": "+",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "+",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": "dec-button",
            "nodeType": "host",
            "props": Object {
              "children": "-",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "-",
            "type": "button",
          },
        ],
        "type": Symbol(react.fragment),
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "£",
          "0.00",
          <UNDEFINED>
            <button
              onClick={[Function]}
            >
              +
            </button>
            <button
              onClick={[Function]}
            >
              -
            </button>
          </UNDEFINED>,
        ],
      },
      "ref": null,
      "rendered": Array [
        "£",
        "0.00",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <button
                onClick={[Function]}
              >
                +
              </button>,
              <button
                onClick={[Function]}
              >
                -
              </button>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "inc-button",
              "nodeType": "host",
              "props": Object {
                "children": "+",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "+",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": "dec-button",
              "nodeType": "host",
              "props": Object {
                "children": "-",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "-",
              "type": "button",
            },
          ],
          "type": Symbol(react.fragment),
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Wallet Renders loading message when loading is true 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Wallet
    loading={true}
    onClick={undefined}
    value={0}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Please wait",
    },
    "ref": null,
    "rendered": "Please wait",
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Please wait",
      },
      "ref": null,
      "rendered": "Please wait",
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Wallet Renders value when passed 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Wallet
    loading={false}
    onClick={undefined}
    value={5}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        "£",
        "5.00",
        "",
      ],
    },
    "ref": null,
    "rendered": Array [
      "£",
      "5.00",
      "",
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "£",
          "5.00",
          "",
        ],
      },
      "ref": null,
      "rendered": Array [
        "£",
        "5.00",
        "",
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Wallet Renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Wallet
    loading={false}
    onClick={undefined}
    value={0}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        "£",
        "0.00",
        "",
      ],
    },
    "ref": null,
    "rendered": Array [
      "£",
      "0.00",
      "",
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "£",
          "0.00",
          "",
        ],
      },
      "ref": null,
      "rendered": Array [
        "£",
        "0.00",
        "",
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
